### PR TITLE
shop_and_product_details

### LINKED ISSUE #114: Product details django template
https://github.com/egypt-shops/xshop-docs/blob/develop/detailed_design/user_stories/DEC_story/DEC_mindmap.png

https://github.com/egypt-shops/xshop-docs/blob/develop/detailed_design/user_stories/DEC_story/DEC_wireframe.png

### LINKED ISSUE #143: design the product's details page

### LINKED ISSUE #166: Product detail django template
closes #143

### PR DESCRIPTION
**Pull request by HMMK99**

### CODE REVIEW

**Comment by HMMK99** [comment]
Screenshot from 2021-05-04 10-18-47
Screenshot from 2021-05-04 10-18-53
Screenshot from 2021-05-04 10-19-20

File: xshop/shops/views.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from django.shortcuts import render
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic import ListView
+
+from xshop.shops.models import Shop
+from xshop.products.models import Product
+
+
+class ShopView(LoginRequiredMixin, ListView):
+    def post(self, request, *args, **kwargs):  # class based view def post, def get
```
Comment (mmostafa74) [inline xshop/shops/views.py]: why?

File: xshop/shops/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from django.test import Client, TestCase, tag
+from django.urls import reverse
+from model_bakery import baker
+
+from xshop.shops.models import Shop
+from xshop.users.models import User
+
+
+@tag("cartview")
+class CartTests(TestCase):
```
Comment (mmostafa74) [inline xshop/shops/tests/test_views.py]: Rename CartTests to ShopDetailViewTests

File: xshop/shops/views.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from django.shortcuts import render
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic import ListView
+
+from xshop.shops.models import Shop
+from xshop.products.models import Product
+
+
+class ShopView(LoginRequiredMixin, ListView):
```
Comment (mmostafa74) [inline xshop/shops/views.py]: ShopDetailView

File: xshop/products/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+from django.test import Client, TestCase, tag
+from django.urls import reverse
+from model_bakery import baker
+
+from xshop.products.models import Product
+from xshop.users.models import User
+
+
+@tag("cartview")
+class CartTests(TestCase):
```
Comment (mmostafa74) [inline xshop/products/tests/test_views.py]: change CartTests

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

File: xshop/products/tests/test_views.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        self.url = reverse(
+            "product:product_details", kwargs={"product_id": self.product.id}
+        )
+
+    def test_get_product_details_not_authenticated(self):
+        resp = self.client.get(self.url)
+
+        self.assertEqual(resp.status_code, 302)
+
+    def test_get_product_details(self):
```
Comment (mmostafa74) [inline xshop/products/tests/test_views.py:38]: assert template used, and content

File: xshop/shops/tests/test_views.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        self.shop = baker.make(Shop, mobile="01222222222")
+        self.client = Client()
+        self.url = reverse("shop:shop_details", kwargs={"shop_id": self.shop.id})
+
+    def test_get_shop_details_not_authenticated(self):
+        resp = self.client.get(self.url)
+
+        self.assertEqual(resp.status_code, 302)
+
+    def test_get_shop_details(self):
```
Comment (mmostafa74) [inline xshop/shops/tests/test_views.py:38]: same

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

**Review by mmostafa74 — APPROVED** [review]

**Pull request merged**